client/src/components/SlackModal/SlackModal.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+
+import { useSelector, useDispatch } from 'react-redux';
+import axios from 'axios';
+import {
+  Box,
+  Button,
+  Typography,
+  Modal,
+  TextField,
+  Paper,
+} from '@material-ui/core';
+
+const SlackMessageTextBox = () => {
+  const dispatch = useDispatch();
+
+  const { slackMessageText } = useSelector(
+    (state) => state.surveyDetailReducer,
+  );
+
+  const handleSlackMessage = (event) => {
+    dispatch({
+      type: 'ADD_SLACK_MESSAGE',
+      payload: event.target.value,
+    });
+  };
+
+  return (
+    <Box my={4} p={4}>
+      <Typography>Enter your message to the survey recipients:</Typography>
+      <TextField
+        data-testid='slack-message-box'
+        autoFocus
+        fullWidth
+        required
+        autoComplete='off'
+        value={slackMessageText}
+        error={slackMessageText.length < 5}
+        helperText={
+          slackMessageText < 5
+            ? 'You must provide a message to the recipients'
+            : ''
+        }
+        margin='normal'
+        key='slack-message-box'
+        name='comment-text-box'
+        onChange={handleSlackMessage}
+      />
+    </Box>
+  );
+};
+
+const SlackModal = () => {
+  const dispatch = useDispatch();
+
+  const { openSlackModal, recipients, slackMessageText, _id } = useSelector(
+    (state) => state.surveyDetailReducer,
+  );
+  const { employeeData } = useSelector((state) => state.employeeTableReducer);
+
+  const recipientsIDs = recipients.map((recipient) => recipient.employeeId);
+
+  const generateLink = (recipientID, surveyIdToDo) => {
+    const link = `localhost:3000/${surveyIdToDo}/${recipientID}`;
```

Comment:
Specifying as `localhost:3000` might have problems in future when it's deployed.
This might be better `window.location.hostname` - https://www.w3schools.com/js/js_window_location.asp

Reply:
okay good to know, i will change that!